src/operator/tensor/util/node_op_util.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file node_op_util.h
+ * \brief abstraction for commonly nnvm::Node operations.
+ */
+#ifndef MXNET_OPERATOR_TENSOR_UTIL_NODE_OP_UTIL_H_
+#define MXNET_OPERATOR_TENSOR_UTIL_NODE_OP_UTIL_H_
+#include <mxnet/base.h>
+#include <string>
+#include <unordered_map>
+#include "../../elemwise_op_common.h"
+
+namespace mxnet {
+namespace util {
+
+class NodeOp {
+ private:
+    const nnvm::NodePtr &dependent_node;
```

Comment:
Why a reference to the ptr?

Reply:
I guess to avoid the copy

Reply:
Yeah it is to avoid the copy.
Even `MakeNode` takes a pointer so thought to avoid a copy.
https://github.com/apache/incubator-mxnet/blob/9410cc4ff6b26723beb95cef39001b245392b840/src/operator/operator_common.h#L362-L380